manim/scene/scene.py

```diff
@@ -702,77 +720,12 @@ def compile_method(state):
 
         return animations
 
-    def begin_animations(self, animations):
-        """
-        This method begins the list of animations that is passed,
-        and adds any mobjects involved (if not already present)
-        to the scene again.
-
-        Parameters
-        ----------
-        animations : list
-            List of involved animations.
-
-        """
-        for animation in animations:
-            # Begin animation
-            animation.begin()
-
-    def progress_through_animations(self):
-        """
-        This method progresses through each animation
-        in the list passed and and updates the frames as required.
-        """
-        for t in self.get_animation_time_progression(self.animations):
-            self.update_animation_to_time(t)
-            self.renderer.update_frame(self, self.moving_mobjects, self.static_image)
-            self.renderer.add_frame(self.renderer.get_frame())
-
-    def update_animation_to_time(self, t):
-        """
-        Updates the current animation to the specified time.
-
-        Parameters
-        ----------
-        t : int
-            Offset from the start of the animation to which to update the current
-            animation.
-        """
-        dt = t - self.last_t
-        self.last_t = t
-        for animation in self.animations:
-            animation.update_mobjects(dt)
-            alpha = t / animation.run_time
-            animation.interpolate(alpha)
-        self.update_mobjects(dt)
-
-    def finish_animations(self, animations):
-        """
-        This function cleans up after the end
-        of each animation in the passed list.
-
-        Parameters
-        ----------
-        animations : list
-            list of animations to finish.
-        """
-        for animation in animations:
-            animation.finish()
-            animation.clean_up_from_scene(self)
-        # TODO: This is only used in one place and should probably be removed.
-        self.mobjects_from_last_animation = [anim.mobject for anim in animations]
-        if file_writer_config["skip_animations"]:
-            # TODO, run this call in for each animation?
-            self.update_mobjects(self.get_run_time(animations))
```

Comment:
Just a quick question about these removed lines. 
Suppose that animation n is skipped, but not animation n+1, and animation n+1 uses some stuff that were supposedly updated by animation n, in any way. Will this work without these lines? The mobject that was supposed to be updated in animation n wouldn't have been.

Reply:
Manim handles animation skipping by setting the time progression used in the animation loop to one which holds a single value, the length of the animations being played.

https://github.com/ManimCommunity/manim/blob/3b6519a58678eb84bb31f550a0a4734df58bded3/manim/scene/scene.py#L562-L566

so the relevant mobjects will have already been updated by the time they reach this point.